diff --git a/modules/image/image.field.inc b/modules/image/image.field.inc
index 1be1583..c2f7a37 100644
--- a/modules/image/image.field.inc
+++ b/modules/image/image.field.inc
@@ -75,6 +75,8 @@ function image_field_instance_settings_form($field, $instance) {
   // Use the file field instance settings form as a basis.
   $form = file_field_instance_settings_form($field, $instance);
 
+  // Extend allowed extensions description
+  $form['file_extensions']['#description'] .= '<br/>' . t('Be aware that .png .gif .jpg and .jpeg are the only supported extensions. You are allowed to specify other extensions, but you have to expect that these cannot be output properly with the standard image toolkit.');
   // Add maximum and minimum resolution settings.
   $max_resolution = explode('x', $settings['max_resolution']) + array('', '');
   $form['max_resolution'] = array(
@@ -339,10 +341,9 @@ function image_field_widget_form(&$form, &$form_state, $field, $instance, $langc
     }
 
     // If not using custom extension validation, ensure this is an image.
-    $supported_extensions = array('png', 'gif', 'jpg', 'jpeg');
-    $extensions = isset($elements[$delta]['#upload_validators']['file_validate_extensions'][0]) ? $elements[$delta]['#upload_validators']['file_validate_extensions'][0] : implode(' ', $supported_extensions);
-    $extensions = array_intersect(explode(' ', $extensions), $supported_extensions);
-    $elements[$delta]['#upload_validators']['file_validate_extensions'][0] = implode(' ', $extensions);
+    $supported_extensions = 'png gif jpg jpeg';
+    $extensions = isset($elements[$delta]['#upload_validators']['file_validate_extensions'][0]) ? $elements[$delta]['#upload_validators']['file_validate_extensions'][0] : $supported_extensions;
+    $elements[$delta]['#upload_validators']['file_validate_extensions'][0] = $extensions;
 
     // Add all extra functionality provided by the image widget.
     $elements[$delta]['#process'][] = 'image_field_widget_process';